<launch>
  <arg name="FIXED_FRAME" default="/odom" />  <!-- /imu_floor -->
  <arg name="TARGET_FRAME" default="/head_hokuyo_frame" />
  <arg name="USE_GROUND_FILTER" default="false" />

  <!-- start tilting -->
  <node pkg="laser_filters" type="scan_to_cloud_filter_chain"
        name="scan_cloud_filter_to_octomap"
        output="screen" >
    <remap from="scan" to="/multisense_sl/laser/scan"/>
    <remap from="cloud_filtered" to="~cloud_filtered"/>
    <param name="target_frame" value="$(arg TARGET_FRAME)" />
    <rosparam command="load" file="$(find hrpsys_gazebo_atlas)/config/atlas_scan_to_cloud_long_range.yaml" />
  </node>

  <node pkg="octomap_server" type="octomap_server_node" name="octomap_filter_server">
    <remap from="cloud_in" to="scan_cloud_filter_to_octomap/cloud_filtered" />

    <param name="resolution" value="0.20" />
    <param name="sensor_model/max_range" value="15.0" />
    <param name="frame_id" value="$(arg FIXED_FRAME)" />
    <!-- for filter ground -->
    <param name="filter_ground" value="$(arg USE_GROUND_FILTER)" />
    <param name="base_frame_id" value="/l_foot" /> <!-- imu_floor is not floor -->
    <param name="ground_filter/distance" value="0.06" />
    <param name="ground_filter/angle"    value="0.12" />
    <param name="ground_filter/plane_distance" value="0.12" />
  </node>

</launch>
